src/iceberg/manifest_entry.h

```diff
@@ -77,9 +77,8 @@ struct ICEBERG_EXPORT DataFile {
   FileFormatType file_format;
   /// Field id: 102
   /// Partition data tuple, schema based on the partition spec output using partition
-  /// field ids for the struct field ids
-  /// TODO(zhjwpku): use StructLike to represent partition data tuple
-  std::any partition;
+  /// field ids
+  std::vector<Literal> partition;
```

Comment:
This is ok to me, and should we extract a `Literals` or `StructLike` type using `Vector<Literal>`, or just leave it here?